tests/conftest.py

```diff
@@ -77,13 +80,31 @@ def hass_storage():
 
 
 @pytest.fixture
-def hass(loop, hass_storage):
+def hass(loop, hass_storage, request):
     """Fixture to provide a test instance of Home Assistant."""
+
+    def exc_handle(loop, context):
+        """Handle exceptions by rethrowing them, which will fail the test."""
+        exceptions.append(context["exception"])
+        orig_exception_handler(loop, context)
+
+    exceptions = []
     hass = loop.run_until_complete(async_test_home_assistant(loop))
+    orig_exception_handler = loop.get_exception_handler()
+    loop.set_exception_handler(exc_handle)
 
     yield hass
 
     loop.run_until_complete(hass.async_stop(force=True))
+    for ex in exceptions:
+        if (
+            request.module.__name__,
+            request.function.__name__,
+        ) in IGNORE_UNCAUGHT_EXCEPTIONS:
+            continue
+        if isinstance(ex, ServiceNotFound):
+            continue
```

Comment:
I wonder if we can reduce this by setting up persistent notification, which seems to be causing most errors. But that is for future us, this is already a great improvement.

Reply:
the problem is that it happens for components that don't use persistent notifications directly (like mine, Dynalite...). I plan to take a look at this as part of the exception reduction, but i have a bit less than 200 exceptions today in the file. If i remove this line it becomes 1600.

Reply:
We could consider to set it up as part of the hass fixture.

Reply:
Right now we ignore all service not found errors, I bet that we're missing some real ones too. But let's keep this for the future.

Reply:
i believe that setting it for all the tests will increase the test time significantly. We can add mock service calls for the persistent notifications in the hass fixture and will likely catch the same errors since persistent notifications are always supposed to be set up, right?